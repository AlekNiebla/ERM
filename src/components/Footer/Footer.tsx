import * as React from "react";

const Footer = () => <span>Footer</span>;

export default Footer;
