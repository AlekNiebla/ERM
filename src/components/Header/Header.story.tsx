import * as React from "react";
import { storiesOf } from "@storybook/react";
import Header from "./Header";

storiesOf("Components / Header", module)
  .add("default", () => <Header />)
  .add("with content", () => (
    <Header>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Animi nostrum id
      voluptates nisi ut voluptate voluptatem repellat eos culpa molestiae ipsum
      omnis eaque iusto quisquam, dolorem accusantium earum officiis. Sunt.
    </Header>
  ));
