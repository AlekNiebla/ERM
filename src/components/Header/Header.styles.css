.main {
  background-color: var(--color-gray-lightest);
  display: flex;
  flex-direction: column;
  border-bottom: 1px solid var(--color-gray-lighter);
}
.content {
  padding: var(--step-5);
}
