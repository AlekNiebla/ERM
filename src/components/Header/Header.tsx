import * as React from "react";
// components
import LogoBar from "./../LogoBar/LogoBar";

interface IHeader {
  actionItem?: React.ReactNode;
  children?: React.ReactNode;
}

const Header = (props: IHeader) => (
  <div>
    <LogoBar />
    {props.children}
  </div>
);

export default Header;
