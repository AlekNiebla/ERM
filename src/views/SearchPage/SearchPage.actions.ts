import {
  MovieListQuery,
  IMovie,
  ActionTypeMap,
  ThunkAction
  // StoreDispatch,
  // PayloadAction
} from "Types";
import movieService from "Services/movieService";
import globalActionTypes, { toggleFetchStatus } from "Root/global.actions";

const searchActionTypes: ActionTypeMap = {
  getMovieList: "GET_MOVIE_LIST_QUERY",
  getMovieListResponse: "GET_MOVIE_LIST_RESPONSE",
  getMovieListError: "GET_MOVIE_LIST_ERROR",
  // getMovieDetails: "GET_MOVIE_DETAILS_QUERY",
  // getMovieDetailsResponse: "GET_MOVIE_DETAILS_RESPONSE",
  // getMovieDetailsError: "GET_MOVIE_DETAILS_ERROR",
  movieListSorting: "MOVIE_LIST_SORT"
};

export default searchActionTypes;

// helperFunction dispatching response actions
export const getMovie = async (dispatch: any, query: MovieListQuery) => {
  const result: Error | IMovie[] = await movieService.getMovies(query);

  dispatch(toggleFetchStatus(false));
  dispatch({
    type: Array.isArray(result)
      ? searchActionTypes.getMovieListResponse
      : globalActionTypes.fetchError,
    payload: result
  });
};

// thunk action
export const movieSearch: ThunkAction<
  MovieListQuery,
  boolean
> = query => dispatch => {
  dispatch(toggleFetchStatus(true));
  getMovie(dispatch, query);

  return {
    type: searchActionTypes.getMovieList,
    payload: query
  };
};
