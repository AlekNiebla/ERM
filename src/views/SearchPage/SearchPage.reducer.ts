import { ActionType } from "AppRoot/types";
import searchActionTypes from "./SearchPage.actions";
import { IMovie } from "Components/MovieGrid/MovieGrid";

type SearchPageReducerState = {
  readonly query: false | string;
  readonly movies: false | IMovie[];
};

const initialState: SearchPageReducerState = {
  query: false,
  movies: false
};

const searchPageReducer = (state = initialState, action: ActionType) => {
  switch (action.type) {
    case searchActionTypes.getMovieList:
      return {
        ...state,
        query: action.payload
      };
    case searchActionTypes.getMovieListResponse:
      return {
        ...state,
        movies: action.payload
      };
    case searchActionTypes.getMovieListError:
    default:
      return state;
  }
};

export default searchPageReducer;
