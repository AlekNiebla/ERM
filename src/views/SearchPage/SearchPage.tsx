import * as React from "react";

const SearchPage = () => <span>SearchPage</span>;

export default SearchPage;
