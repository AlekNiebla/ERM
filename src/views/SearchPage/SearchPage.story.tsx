import * as React from "react";
import { storiesOf } from "@storybook/react";
import SearchPage from "./SearchPage";

storiesOf("Views / SearchPage", module).add("default", () => <SearchPage />);
