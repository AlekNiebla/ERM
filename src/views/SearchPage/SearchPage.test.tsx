import * as React from "react";
import * as puppeteer from "puppeteer";
import { shallow } from "enzyme";
const path = require("path");

// components
import SearchPage from "./SearchPage";
import Header from "./../../components/Header/Header";
import Footer from "./../../components/Footer/Footer";

describe("SearchPage component...", () => {
  it("... renders without errors", () => {
    const wrapper = shallow(<SearchPage />);

    expect(wrapper.find(SearchPage)).toBeTruthy();
    expect(wrapper.find(Header)).toBeTruthy();
    expect(wrapper.find(Footer)).toBeTruthy();
  });
  it("... matches snapshot", () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper).toMatchSnapshot();
  });

  // INTERGRATION TESTS
  describe("... passes integration e2e tests ...", () => {
    it("... accepts input", async () => {
      const browser = await puppeteer.launch();
      const page = await browser.newPage();

      await page.goto(path.resolve(__dirname, "./../../../dist/index.html"));
      await page.type("input", "test movie title");
      await page.click("button");
      const text = await page.$eval("input", el => el.getAttribute("value"));

      expect(text).toBe("test movie title");
    });
  });
});
