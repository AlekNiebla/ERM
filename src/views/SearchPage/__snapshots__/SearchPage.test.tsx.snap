// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage component... ... matches snapshot 1`] = `ShallowWrapper {}`;
